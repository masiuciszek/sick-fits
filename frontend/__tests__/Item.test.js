import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ItemComponent from '../components/Item';

const fakeItem = {
  id: 'asdasd212',
  title: 'cool shoes!',
  price: 3323,
  description: 'love this shoes',
  image: 'show.jpg',
  largeImage: 'showxxl.jpg',
};

describe('<Item/>', () => {
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  // it('renders adn displays properly', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //   // console.log(wrapper.debug());
  //   const priceTag = wrapper.find('PriceTag');
  //   expect(priceTag.children().text()).toBe('Â£33.23');
  //   expect(wrapper.find('Title a').text()).toBe(fakeItem.title);
  //   const img = wrapper.find('img');
  //   expect(img.props().src).toBe(fakeItem.image);
  //   expect(img.props().alt).toBe(fakeItem.title);
  // });
  // it('renders out the buttons properly', () => {
  //   const wrapper = shallow(<ItemComponent item={fakeItem} />);
  //   const buttonList = wrapper.find('.buttonList');
  //   expect(buttonList.children()).toHaveLength(3);
  //   expect(buttonList.find('Link')).toHaveLength(1);
  // });
});
