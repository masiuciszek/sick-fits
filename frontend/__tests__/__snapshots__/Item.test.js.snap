// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="cool shoes!"
    src="show.jpg"
  />
  <Title>
    <Link
      as="/item/cool shoes!"
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "asdasd212",
          },
        }
      }
    >
      <a>
        cool shoes!
      </a>
    </Link>
  </Title>
  <PriceTag>
    Â£33.23
  </PriceTag>
  <p>
    love this shoes
  </p>
  <div
    className="buttonList"
  >
    <Link
      as="/update/cool shoes!"
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "asdasd212",
          },
        }
      }
    >
      <a>
         Edit ðŸ–Œ 
      </a>
    </Link>
    <AddToCart
      id="asdasd212"
    />
    <DeleteItem
      icon="ðŸ—‘"
      id="asdasd212"
      text="Delete Item"
    />
  </div>
</ItemStyles__Item>
`;
